### PR TITLE
test: add tap output

### PR DESCRIPTION
**Pull request by tjfontaine**
Given UV_TAP_OUTPUT being set, test result output should use TAP formatting

TODO: Windows implementation

### CODE REVIEW

File: test/runner-unix.c [hunk 1]
```diff
@@ -49,6 +51,12 @@ void platform_init(int argc, char **argv) {
     exit(1);
   }
 
+  if (tap != NULL && atoi(tap) > 0) {
+    tap_output = 1;
+  } else {
+    tap_output = 0;
+  }
```
Comment (bnoordhuis) [inline test/runner-unix.c]:
``` c
tap_output = (tap != NULL && atoi(tap) > 0);
```

?

File: test/runner-unix.c [hunk 2]
```diff
@@ -38,10 +38,12 @@
 #include <sys/select.h>
 #include <pthread.h>
 
+int tap_output;
```
Comment (bnoordhuis) [inline test/runner-unix.c]: Style: Two newlines between variables and functions.

File: test/runner-unix.c [hunk 3]
```diff
@@ -263,17 +271,26 @@ int process_copy_output(process_info_t *p, int fd) {
...
+    off = 0;
+  }
 
-  while ((nread = read(fileno(p->stdout_file), buf, 1024)) > 0) {
-    nwritten = write(fd, buf, nread);
-    /* TODO: what if write doesn't write the whole buffer... */
+  /* TODO: what if the line is longer than buf */
+  while (fgets(buf + off, sizeof(buf) - off, p->stdout_file) != NULL) {
+   /* TODO: what if write doesn't write the whole buffer... */
+    nwritten = write(fd, buf, strlen(buf));
```
Comment (bnoordhuis) [inline test/runner-unix.c]: Why not:

``` c
nwritten = 0;

if (tap_output)
  nwritten += write(fd, "#", 1);

nwritten += write(fd, buf, strlen(buf));
```

File: test/runner.c [hunk 4]
```diff
@@ -243,10 +249,14 @@ int run_test(const char* test, int timeout, int benchmark_output) {
...
-      LOGF("\n`%s` failed: %s\n", test, errmsg);
+    if (tap_output) {
+      LOGF("not ok %d - %s\n#", tcount, test);
     } else {
-      LOGF("\n");
+      if (status != 0) {
+        LOGF("\n`%s` failed: %s\n", test, errmsg);
+      } else {
+        LOGF("\n");
+      }
```
Comment (bnoordhuis) [inline test/runner.c]: Fold into outer if/else block (i.e. `else if (...)`).

File: test/runner.c [hunk 5]
```diff
@@ -117,7 +123,7 @@ int run_tests(int timeout, int benchmark_output) {
 }
 
 
-int run_test(const char* test, int timeout, int benchmark_output) {
+int run_test(const char* test, int timeout, int benchmark_output, int tcount) {
```
Comment (bnoordhuis) [inline test/runner.c]: s/tcount/test_count/ ?

**Comment by tjfontaine** [comment]
ok force pushed, all those suggestions certainly make for much more elegant code :)

**Comment by tjfontaine** [comment]
force pushed with more style changes

**Comment by bnoordhuis** [comment]
Landed with some final fix-ups in bfe269b.

**Pull request closed without merging**